var/spack/repos/builtin/packages/py-genshi/package.py

```diff
@@ -9,10 +9,10 @@
 class PyGenshi(PythonPackage):
     """Python toolkit for generation of output for the web"""
     homepage = "https://genshi.edgewall.org/"
-    url      = "http://ftp.edgewall.com/pub/genshi/Genshi-0.7.tar.gz"
+    url = "https://github.com/edgewall/genshi/archive/refs/tags/0.7.7.tar.gz"
```

Comment:
Can we download from PyPI instead? https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github

Reply:
Thank you for pointing this out. Using pypi would make the spack package robust to exactly this type of tarball hosting change.  I checked that the 3 older versions are all also hosted on pypi.